pkg/webhook/kubernetes/tls.go

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright (c) 2012-2019 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package webhook_k8s
+
+import (
+	"context"
+
+	"github.com/devfile/devworkspace-operator/pkg/webhook/service"
+
+	"github.com/devfile/devworkspace-operator/pkg/kubernetes/tls"
+	"github.com/devfile/devworkspace-operator/webhook/server"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var log = logf.Log.WithName("webhook-k8s")
+
+const (
+	TLSSelfSignedCertificateSecretName = "devworkspace-self-signed-certificate"
+)
+
+// SetupSecureService handles TLS secrets required for deployment on Kubernetes.
+func SetupSecureService(client crclient.Client, ctx context.Context, namespace string) error {
+	devworkspaceSecret := &corev1.Secret{}
+
+	err := client.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: server.WebhookServerTLSSecretName}, devworkspaceSecret)
+	if !errors.IsNotFound(err) {
```

Comment:
I constantly get `Error getting TLS secret ` and interesting part that the whole process does not fail because current err is nil :-D
I'm proving proposal as usual )
https://github.com/devfile/devworkspace-operator/compare/kubernetes-cert-job...sleshchenko:kubernetes-cert-job-proposal-3?expand=1